AddMatch.tsx

```diff
@@ -229,11 +234,12 @@ function checkData(players: PlayerData[], selectedPositions: Selection) {
   }
   return true
 }
-function send(
+async function send(
   selectedPositions: Selection,
   setSelectedPositions: (sel: Selection) => void,
   matchesCount: number,
-  setMatchesCount: (count: number) => void
+  setMatchesCount: (count: number) => void,
+  setSaving: (saving: boolean) => void
```

Comment:
Creo que lo único que cambiaría es que `send` solamente guarde la info y no reciba setters y esas cosas, y eso se haría de afuera:

```tsx
setSaving(true)
send(blah).then(() => { // (o con try/await/finally)
  setSelectedPositions({})
  setMatchesCount(matchesCount + 1)
}).finally(() => setSaving(false));
```

Reply:
Ok, pero siendo varias cosas que hacer, si quisiera que no me quede dentro del JSONX ese código, ¿cómo tendría que hacerlo?

Reply:
Bah, capaz yo lo quiero sacar de ahí pero lo usual es dejarlo ahí. Ustedes son los que saben cómo se usa.

Reply:
Se, totalmente. Si querés simplemente sacarlo del JSX, podés más arriba en el componente crear una función y después ya se la pasás armada al `onPress` como para que quede más limpio de leer.

Si nos ponemos a hilar más fino, no sé si tengo una _posta_ (@holden-caulfield?), algo que se me ocurre para simplificar podría ser:

* Un `AsyncButton` que tiene una API parecida a ese `TouchableHighlight` pero que internamente se guarda el flag de si está in progress y lo setea como corresponde
* Un hook como `useMatch` o algo así que devuelva funciones un poco más digeridas para manejar un match
* Quizás como para simplificar extra, extraer un `useCounter` que devuelve el valor del contador y una función para incrementarlo

Usando todo, queda algo como

```tsx
const [matchesCount, incMatchesCount] = useCounter();
const { selectedPositions, setPosition, resetPositions } = useMatch();

// ...

return (
  <AsyncButton
    onPress={() =>
      send(selectedPositions, matchesCount)
        .then(incMatchesCount).then(resetPositions)
    }
  >
    Agregar
  </AsyncButton>
)
```

Reply:
A ver qué te parece así:
```
  // ...
  const waitingAction = (action: Promise<any>) => {
    setSaving(true)
    action
      .then(() => {
        setMatchesCount(matchesCount + 1)
        setSelectedPositions({})
      })
      .catch((e) => Alert.alert('Error guardando partida', String(e)))
      .finally(() => setSaving(false))
  }
  return (
  <TouchableHighlight
    onPress={() => waitingAction(send(selectedPositions))}>
    Agregar
  </TouchableHighlight >
)
```
Lo que no me cabe del AsyncButton es que el estado es interno y no puedo usarlo para deshabilitar otras cosas mientras espera, como por ejemplo los botones de las posiciones (que dicho sea de paso agrego eso que colgué en hacerlo).

Reply:
Ahí mandé cambios y creé un componente RoundedButton con algo de estilos y algo de comportamiento. Fíjate si así te gusta. 🤞

Reply:
De una 👍  aunque esto es cierto también

> Lo que no me cabe del AsyncButton es que el estado es interno y no puedo usarlo para deshabilitar otras cosas mientras espera

Así que quizás fue una sugerencia medio pedorra :P